src/state/StateClient.cpp

```diff
@@ -1,138 +1,193 @@
 #include <faabric/state/StateClient.h>
```

Comment:
As discussed at the description of the PR, the state API is blocking. Thus, even if messages expect no response, they block until receiving an empty response from the server.